src/vivarium/examples/boids/forces.py

```diff
@@ -47,6 +47,8 @@ def setup(self, builder: Builder) -> None:
     def apply_force(self, index: pd.Index[int], acceleration: pd.DataFrame) -> pd.DataFrame:
         neighbors = self.neighbors(index)
         pop = self.population_view.get(index)
+        if not isinstance(neighbors, pd.Series):
```

Comment:
This is because it could be a dataframe? Do we need to narrow our focus even more to have a `SeriesAttributePipeline` and a `DataFrameAttributePipeline`?

Reply:
Well, kinda but I think of it in the opposite direction. it's just that the next call `self._get_pairs(neighbors, pop)` type hints neighbors as being a series.